### PR TITLE
pthread/cond: Use clock_id from condattr

### PR DESCRIPTION
**Pull request by agkaminski**
JIRA: RTOS-639

## Description

Fixes https://github.com/phoenix-rtos/phoenix-rtos-project/issues/876

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Comment by agkaminski** [comment]
@lukileczo Did I understood correctly how it's supposed to work? I'm not so sure myself

File: include/sys/types.h [hunk 1]
```diff
@@ -75,6 +75,7 @@ typedef struct pthread_mutexattr_t {
 typedef struct {
 	handle_t condh;
 	volatile int initialized;
+	clockid_t clock_id;
```
Comment (lukileczo) [inline include/sys/types.h:78]: I don't think we should add `clockid_t` here, without `condattr` it should just use some default value

Comment (lukileczo) [reply include/sys/types.h:78]: Ok nvm, I misunderstood this a little bit

**Review by lukileczo — DISMISSED** [review]

File: pthread/pthread.c [hunk 2]
```diff
@@ -845,6 +850,8 @@ int pthread_cond_init(pthread_cond_t *__restrict cond, const pthread_condattr_t
 {
 	int err;
 
+	cond->clock_id = attr->clock_id;
```
Comment (lukileczo) [inline pthread/pthread.c:853]: `attr` can be `NULL`, then `cond->clock_id` should have default value

**Review by lukileczo — CHANGES_REQUESTED** [review]

File: pthread/pthread.c (hunk 2 above)
Comment (anglov) [inline pthread/pthread.c:853]: attr can be null, and then default should be used.

**Review by anglov — COMMENTED** [review]
Thanks for this change. It makes easier to write portable code and will secure from weird errors if potential changes of timers will be made in the future. 

One small remark

File: pthread/pthread.c (hunk 2 above)
Comment (agkaminski) [reply pthread/pthread.c]: Thx, fixed

**Review by lukileczo — APPROVED** [review]

**Comment by nalajcie** [comment]
pthread tests are now failing?

image

**Pull request merged**